invenio_app_ils/document_requests/views.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio App ILS Document Requests views."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint
+from invenio_db import db
+from invenio_records_rest.utils import obj_or_import_string
+from invenio_records_rest.views import pass_record
+from invenio_rest import ContentNegotiatedMethodView
+from invenio_rest.errors import FieldError
+
+from invenio_app_ils.errors import DocumentRequestError
+from invenio_app_ils.mail.tasks import send_document_request_status_mail
+from invenio_app_ils.permissions import need_permissions
+from invenio_app_ils.proxies import current_app_ils
+
+from .api import DOCUMENT_REQUEST_PID_TYPE
+from .loaders import document_request_accept_loader, \
+    document_request_reject_loader
+
+
+def create_document_request_action_blueprint(app):
+    """Create document request action blueprint."""
+    blueprint = Blueprint("ils_document_request", __name__, url_prefix="")
+
+    endpoints = app.config.get("RECORDS_REST_ENDPOINTS", [])
+    options = endpoints.get(DOCUMENT_REQUEST_PID_TYPE, {})
+    default_media_type = options.get("default_media_type", "")
+    rec_serializers = options.get("record_serializers", {})
+    serializers = {
+        mime: obj_or_import_string(func)
+        for mime, func in rec_serializers.items()
+    }
+
+    pending_view = PendingRequestResource.as_view(
+        PendingRequestResource.view_name.format(DOCUMENT_REQUEST_PID_TYPE),
+        serializers=serializers,
+        default_media_type=default_media_type,
+    )
+    blueprint.add_url_rule(
+        "{0}/pending".format(options["item_route"]),
+        view_func=pending_view,
+        methods=["POST"],
+    )
+
+    accept_view = AcceptRequestResource.as_view(
+        AcceptRequestResource.view_name.format(DOCUMENT_REQUEST_PID_TYPE),
+        serializers=serializers,
+        default_media_type=default_media_type,
+        ctx=dict(loader=document_request_accept_loader),
+    )
+    blueprint.add_url_rule(
+        "{0}/accept".format(options["item_route"]),
+        view_func=accept_view,
+        methods=["POST"],
+    )
+
+    reject_view = RejectRequestResource.as_view(
+        RejectRequestResource.view_name.format(DOCUMENT_REQUEST_PID_TYPE),
+        serializers=serializers,
+        default_media_type=default_media_type,
+        ctx=dict(loader=document_request_reject_loader),
+    )
+    blueprint.add_url_rule(
+        "{0}/reject".format(options["item_route"]),
+        view_func=reject_view,
+        methods=["POST"],
+    )
+
+    return blueprint
+
+
+class PendingRequestResource(ContentNegotiatedMethodView):
```

Comment:
I didn't use a loader because at least until now, it doesn't have any data, will add the loader if we need to pass extra information when Ill or Acq come in play.